Cirrus: Give more memory to golangci-lint

cevich opened the pull request:
/kind failing-test 

#### What this PR does / why we need it:

Increase memory available to `golangci-lint` so that it isn't randomly OOMkilled.

#### How to verify it

Run the automated tests

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

None

#### Does this PR introduce a user-facing change?

None

cevich commented:
```
Failed to start: Invalid value for field 'resource.machineType': 'https://www.googleapis.com/compute/v1/projects/buildah/zones/us-central1-c/machineTypes/custom-2-16384'. Memory size for 2 vCPU instance should be between 2048MiB and 13312MiB, while 16384MiB is requested
```

TomSweeneyRedHat commented on .cirrus.yml:
```diff
@@ -92,7 +92,11 @@ gce_instance:
     skip: $CIRRUS_PR == ''
 
-    timeout_in: 30m
+    timeout_in: 10m
```
Just curious why you lowered the time?

cevich replied:
To avoid leaving a big-expensive VM hanging around any longer than necessary.  For this task, if it takes longer than a minute or three, it's likely somethings very wrong and it needs to timeout.

cevich replied:
Do you think this is worthy of a comment in the code?

TomSweeneyRedHat replied:
Wouldn't hurt, but I'd add it later.  I'd like to get this in sooner, especially given the green test buttons.

TomSweeneyRedHat commented:
One question otherwise 
LGTM
and happy green test buttons.  (Maybe never question success @TomSweeneyRedHat ?)

TomSweeneyRedHat commented:
@nalind @dwalsh PTAL

nalind commented:
I mean, it clearly seems to help, but I do wonder if we need to go all the way from 2 vcpus with 4GB of memory to 8 vcpus with 32GB of memory to get that result.  Have we still been seeing the problem with values in between, say 12GB, which is within the bounds of what we can request with a 2 vcpu instance?

rhatdan commented:
If @nalind change works, do we still need this one?

nalind commented:
Well, that was a guess, not based on profiling or any data.

cevich commented:
Mine is a educated guess as well.  Educated based on hitting something similar in podman, and finding the 16-18gig to be about  the sweet-spot.  The gating tests run so quickly, I'm not overly concerned about "overshooting", and the chance of memory needs growing over time seems greater than it reducing.  That was my thought process.

cevich commented:
Rebased, changed to 4CPU/16Gig as maybe a sweet-spot.  Leaving this open for a little while, just in case it's needed (Nalin's 12gig is found to be insufficient).

cevich commented:
We can close this at start of Nov. if there are no further OOM killing instances in gate tests...and If I forget to close it sooner :smile:

rhatdan commented:
/hold

rhatdan commented:
Since we are past halloween, I am going to close this.  Cirrus seems to be working well.

cevich commented:
Thanks Dan.
